src/components/Genomes/Browser/index.tsx

```diff
@@ -32,6 +32,48 @@ const GenomeBrowser: React.FC = () => {
     return response.ok;
   }, [virifyGffUrl]);
 
+  const resolveQueryParameters = (browser, optionTrackName) => {
+    const currentUrl = new URL(window.location.href);
+    const featureId = currentUrl.searchParams.get('feature-id');
+    const contigId = currentUrl.searchParams.get('contig-id');
+    const selectedTrackColor = currentUrl.searchParams.get(
+      'functional-annotation'
+    );
+    if (featureId) {
+      browser.search(featureId);
+    }
+    if (contigId) {
+      browser.search(contigId);
+    }
+    if (selectedTrackColor) {
+      const trackColorBy = {
+        label: selectedTrackColor,
+        value: selectedTrackColor,
+      };
+      setTrackColorBys({
+        ...trackColorBys,
+        [optionTrackName]: trackColorBy,
+      });
+    }
+  };
+
+  const updateQueryParams = (key, value) => {
+    const currentUrl = new URL(window.location.href);
+    currentUrl.searchParams.set(key, value);
+    const updatedUrl = currentUrl.toString();
+    window.history.replaceState(null, null, updatedUrl);
```

Comment:
This just my gut feeling, maybe we need to sort of debounce this. If a user is exploring the genome viewer, changing contigs and features very quickly then this will sort of highjack the browser history.

Should we add delay here.. as to only record the latest action in a series of rapid user interactions?

Reply:
@mberacochea from what I can see, updating the query params does not have any effect on performance. 
Also the values in the query params just get updated. It does not append any new values. 
But perhaps I haven't fully understood your concerns here. Happy to have a chat about this.

Reply:
Well.. I misread the method from `window.history` that you are using. So need to debounce anything.